node/server.js

```diff
@@ -5154,6 +5125,135 @@ function init_io() {
+							for (const [locationsMapKey, locationType, locationIndex, range = 120] of gMap.enter.locations) {
+								if (!G.maps[locationsMapKey]) {
+									server_log(`G.maps.${locationsMapKey} does not exist`);
+									continue;
+								}
+
+								if (!G.maps[locationsMapKey][locationType]) {
+									server_log(`G.maps.${locationsMapKey}.${locationType} does not exist`);
+									continue;
+								}
+
+								if (!G.maps[locationsMapKey][locationType][locationIndex]) {
+									server_log(`G.maps.${locationsMapKey}.${locationType}.${locationIndex} does not exist`);
+									continue;
+								}
+
+								const location = G.maps[locationsMapKey][locationType][locationIndex];
+								const distanceToLocation = simple_distance(player, {
+									in: locationsMapKey,
+									map: locationsMapKey,
+									x: location[0],
+									y: location[1],
+								});
+
+								server_log(
+									`G.maps.${locationsMapKey}.${locationType}.${locationIndex} ${distanceToLocation} <= ${range}`,
+								);
+
+								if (distanceToLocation <= range) {
+									inRange = true;
+									break;
+								}
+							}
+						}
+
+						if (!inRange) {
+							return fail_response("transport_cant_reach");
+						}
+
+						// We check that the instance does not exist so the cost is not paid multiple times.
+						if (gMap.enter.items && !instanceExists) {
+							const itemsToConsume = [];
+
+							const quantityByItem = {};
+							for (const [itemKey, quantity] of gMap.enter.items) {
+								quantityByItem[itemKey] = quantity;
+							}
+
+							for (let i = 0; i < player.items.length; i++) {
+								const item = player.items[i];
+								if (item && quantityByItem[item.name]) {
+									server_log(`${data.place} player has ${item.q} x ${item.name} in slot ${i}`);
+									const gItem = G.items[item.name];
+									if (gItem.s) {
+										const quantity = Math.min(item.q, quantityByItem[item.name]);
+
+										quantityByItem[item.name] -= quantity;
+										server_log(`${data.place} ${data.name} ${quantity} x ${item.name} to be removed`);
+										itemsToConsume.push([i, quantity]);
+									} else {
+										// TODO: handle non stackable items
+										server_log(
+											`${data.place} player has ${item.name} in slot ${i} that is not stackable, how to handle?`,
+										);
+									}
+
+									if (quantityByItem[item.name] == 0) {
+										delete quantityByItem[item.name];
+									}
+								}
+							}
+
+							// validate missing quantities
+							if (Object.keys(quantityByItem).length > 0) {
+								server_log(`${data.place} missing items ${JSON.stringify(quantityByItem)}`);
+								// TODO: tell missing requirements to the client?
+								return fail_response("transport_cant_item");
+							}
+
+							for (const [inventory_index, quantity] of itemsToConsume) {
+								// TODO: will throw an error if quantity is spread over multiple stacks
```

Comment:
I'm not sure this comment is true since you're using `Math.min` to extract the proper quantity from the stack, above.

Reply:
As far as I remember, if you need 2 of an item and you have two items in total but a single one in slot 2 and 3 consume will throw an error because slot 2 does not have a quantity of 2. It's been a minute since I last looked into it, and it will only be relevant if multiple items are required to enter. Currently all cases use a single key.

Reply:
But you are providing `quantity` to `consume` and it is set [here](https://github.com/kaansoral/adventureland/pull/135/files/f56650ec208b983572e0d797a33db6231584975b#diff-a87bee312b94f8153dfcfaabaa0187c4e211f89359516914d0bc0222c1d76ad8R5210) with the proper quantity of the slot. I think it will handle all cases properly.

Reply:
should be solved